csrc/scheduler/matmul_heuristic_plugin_api.h

```diff
@@ -0,0 +1,90 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <memory>
+
+namespace nvfuser {
+
+namespace matmul_heuristic_plugin {
+
+//! This is intended as a minimal interface for enabling matmul heuristics.
+//! In order to plug in your own custom heuristic, create a dynamic library
+//! defining a subclass of KernelConfig, overriding the `configure` method. This
+//! class does not need to be exported from the dll, but you should export a
+//! std::unique_ptr<KernelConfig> makeConfig() function that returns a
```

Comment:
`examples/matmul_heuristic_plugin/matmul_heuristic_plugin.cpp` is `KernelConfig* makeConfig()` instead of using `unique_ptr`. Also, should we declare this function in this header?

Reply:
Thanks. I need to update the comment. I originally made `makeConfig` return a `unique_ptr` in which case it didn't need to be `extern "C"`, but instead now I am returning a raw pointer and converting that to `unique_ptr` on the nvfuser side.

I don't think we can declare the function here since it is the symbol that we will need to load with `dlsym`.